utils/host_info/src/error.rs

```diff
@@ -0,0 +1,64 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use displaydoc::Display;
+use icu_locale_core::ParseError;
+use std::error::Error;
+use std::{ffi::FromVecWithNulError, str::Utf8Error};
+
+/// An error encountered while retrieving the host information
+#[derive(Debug, Display)]
+pub enum HostInfoError {
```

Comment:
thought (non blocking for initial landing): We should eventually consider whether we should split this into fine grained errors. Probably not, most of these are general.

Reply:
Fair enough.